src/main/java/io/pinecone/PineconeControlPlaneClient.java

```diff
@@ -67,14 +65,52 @@ public IndexList listIndexes() {
         return indexList;
     }
 
-    public void deleteIndex(String indexName) {
+    public void deleteIndex(String indexName) throws PineconeException  {
         try {
             manageIndexesApi.deleteIndex(indexName);
         } catch (ApiException apiException) {
             handleApiException(apiException);
         }
     }
 
+    public CollectionModel createCollection(CreateCollectionRequest createCollectionRequest) throws PineconeException {
+        CollectionModel collection = null;
+        try {
+            collection = manageIndexesApi.createCollection(createCollectionRequest);
+        } catch (ApiException apiException) {
+            handleApiException(apiException);
+        }
+        return collection;
+    }
+
+    public CollectionModel describeCollection(String collectionName) throws PineconeException  {
+        CollectionModel collection = null;
+        try {
+            collection = manageIndexesApi.describeCollection(collectionName);
+        } catch (ApiException apiException) {
+            handleApiException(apiException);
+        }
+        return collection;
+    }
+
+    public CollectionList listCollections() throws PineconeException {
+        CollectionList collections = null;
+        try {
+            collections = manageIndexesApi.listCollections();
+        } catch (ApiException apiException) {
+            handleApiException(apiException);
+        }
+        return collections;
+    }
+
+    public void deleteCollection(String collectionName) throws PineconeException {
+        try {
+            manageIndexesApi.deleteCollection(collectionName);
+        } catch (ApiException apiException) {
+            handleApiException(apiException);
+        }
+    }
+
     private void handleApiException(ApiException apiException) {
```

Comment:
We can also add the throws statement here to improve readability

Reply:
Good callout, I missed this one but will add it in.